mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1159,8 +1158,8 @@ def _get_args_for_metrics(self, extra_metric, eval_df):
                     ):
                         eval_fn_args.append(self.other_output_columns[column])
                     elif param.default == inspect.Parameter.empty:
-                        output_column_name = self.evaluator_config.get(
-                            _Y_PREDICTED_OUTPUT_COLUMN_NAME, "output"
+                        output_column_name = (
+                            self.predictions if self.predictions is not None else "output"
```

Comment:
I think lets throw an error instead if the prediction column is not specified for multi output. 
Defaulting to output is confusing to the user in the error message as to where this output column came from?

In this situation, we only add this in the error message:
```
                            "Note that this does not include the output column: "
                            f"'{output_column_name}'\n\n"
```

if there is a `predictions` column.

Reply:
Updated